lib/features/support/interactive_cli.rb

```diff
@@ -0,0 +1,117 @@
+require 'open3'
+
+# Encapsulates a terminal session, retaining state and input streams for interactive tests
+class InteractiveCLI
+  # @!attribute [r] parsed_output
+  #   @return [Array] An array of output strings received from the terminals STDOUT pipe
+  attr_reader :parsed_output
+
+  # @!attribute [r] parsed_error
+  #   @return [Array] An array of error strings received from the terminals STDERR pipe
+  attr_reader :parsed_errors
+
+  # @!attribute [r] last_exit_code
+  #   @return [Number] The exit code of the last terminal command
+  attr_reader :last_exit_code
+
+  # @!attribute [r] pid
+  #   @return [Number] The PID of the running terminal
+  attr_reader :pid
+
+  # @!attribute [r] current_buffer
+  #   @return [String] A string representation of the current output present in the terminal
+  attr_reader :current_buffer
+
+  # Creates an InteractiveCLI instance
+  #
+  # @param environment [Hash] A hash of environment variables
+  # @param terminal [String] A path to the terminal to run, defaults to `/bin/sh`
```

Comment:
I think this would be more accurately a `shell` if the intention is only to execute sh, bash, zsh etc... or maybe just `command` if it can be anything that `Open3` can handle

Reply:
I think `shell` or `terminal` is appropriate as the intent is to present an interactive Terminal.  Making it command could be more accurate, but less descriptive.